server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -680,13 +678,10 @@ protected Symbol visitSubscriptExpression(SubscriptExpression node, ExpressionAn
                     try {
                         Symbol base = fieldProvider.resolveField(qualifiedName, List.of(), operation);
                         if (base instanceof Reference baseRef) {
-                            if (baseRef.valueType().id() == ObjectType.ID) {
-                                if (baseRef.columnPolicy() == ColumnPolicy.STRICT || context.errorOnUnknownObjectKey()) {
-                                    throw new ObjectKeyUnknownException(e.getMessage());
-                                }
-                            } else {
-                                throw e;
+                            if (baseRef.valueType().id() == ObjectType.ID && context.errorOnUnknownObjectKey() == false) {
+                                return Literal.NULL;
```

Comment:
I think I could return Literal.NULL instead of invoking getDynamic. If so, the returned column will be aliased as NULL, aligned with Postgres' ?column?